server/internal/infrastructure/mongo/item_test.go

```diff
@@ -614,3 +614,95 @@ func TestItem_Copy(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, changes, lo.ToPtr(string(wantChanges)))
 }
+
+func TestItem_CountByModel(t *testing.T) {
+	defer util.MockNow(time.Now().Truncate(time.Millisecond).UTC())()
+
+	mid1 := id.NewModelID()
+	mid2 := id.NewModelID()
+	pid1 := id.NewProjectID()
+	pid2 := id.NewProjectID()
+	sid := id.NewSchemaID()
+	sfid := schema.NewFieldID()
+	fs := []*item.Field{item.NewField(sfid, value.TypeBool.Value(true).AsMultiple(), nil)}
+
+	i1 := item.New().NewID().Fields(fs).Schema(sid).Model(mid1).Project(pid1).Thread(id.NewThreadID().Ref()).MustBuild()
+	i2 := item.New().NewID().Fields(fs).Schema(sid).Model(mid1).Project(pid1).Thread(id.NewThreadID().Ref()).MustBuild()
+	i3 := item.New().NewID().Fields(fs).Schema(sid).Model(mid2).Project(pid1).Thread(id.NewThreadID().Ref()).MustBuild()
+	i4 := item.New().NewID().Fields(fs).Schema(sid).Model(mid1).Project(pid2).Thread(id.NewThreadID().Ref()).MustBuild()
```

Comment:
I think this case is not possible, however, you can seed items on different projects to make sure the logic is ok